tachyon/math/polynomials/univariate/univariate_evaluations_ops.h

```diff
@@ -129,6 +133,10 @@ class UnivariateEvaluationsOp {
 
   static Poly& MulInPlace(Poly& self, const F& scalar) {
     std::vector<F>& l_evaluations = self.evaluations_;
+    if (l_evaluations.empty() || scalar.IsOne()) {
```

Comment:
```suggestion
    if (l_evaluations.empty() || scalar.IsZero()) {
```

Reply:
I don't think this is correct -> if scalar is zero, then we should return 0, not self.